### PR TITLE
issue #226 docstring improvement and add IgnoreConditions to __init__.py

### LINKED ISSUE #226: Missing like to IgnoreConditions enum in FlowProject.run() docs
### Description

The link to the `IgnoreConditions` class is not rendering correctly in the docs for `FlowProject.run()`

### To reproduce

Go to https://docs.signac.io/projects/flow/en/latest/api.html#flow.FlowProject.run

### PR DESCRIPTION
**Pull request by zhou-pj**
## Motivation and Context

Resolves #226 

## Types of Changes

- [x] Documentation update
- [x] Bug fix
- [ ] New feature
- [ ] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-flow/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac-flow/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md#code-style) of this project.
- [ ] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [x] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [ ] I have updated the [changelog](https://github.com/glotzerlab/signac-flow/blob/master/changelog.txt).

### CODE REVIEW

File: flow/__init__.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
 from . import scheduling
 from . import errors
 from . import testing
+from .project import IgnoreConditions
```
Comment (zhou-pj) [inline flow/__init__.py]: @vyasr Based on what you mentioned regarding __init__.py last week, I think we need this to allow user to access IgnoreConditions by `from flow import IgnoreConditions`. Is this understanding correct?

Comment (vyasr) [reply flow/__init__.py]: Yes, otherwise you would have to do `from flow.project import IgnoreConditions` or similar.

File: flow/project.py [hunk 2]
```diff
@@ -97,11 +97,13 @@
 
 
 class IgnoreConditions(IntEnum):
-    """IntEnum class for IgnoreConditions settings when checking job eligibility, setting includes:
-    * IgnoreConditions.PRE: ignore pre conditions
-    * IgnoreConditions.POST: ignore post conditions
-    * IgnoreConditions.ALL: ignore all conditions
-    * IgnoreConditions.NONE: check all conditions
+    """IntEnum class for IgnoreConditions settings when checking job eligibility,
```
Comment (vyasr) [inline flow/project.py]:
```suggestion
    """Flags that determine which conditions are used to determine job eligibility.
```

File: flow/project.py [hunk 3]
```diff
@@ -97,11 +97,13 @@
...
+
+    Setting includes:
```
Comment (vyasr) [inline flow/project.py]:
```suggestion
    The options include:
```

File: flow/project.py [hunk 4]
```diff
@@ -1815,21 +1817,19 @@ def run(self, jobs=None, names=None, pretend=False, np=None, timeout=None, num=N
...
                 Users are advised to not rely on a specific execution order, as a
                 substitute for defining the workflow in terms of pre- and post-conditions.
                 However, a specific execution order may be more performant in cases where
                 operations need to access and potentially lock shared resources.
-
         :type order:
             str, callable, or NoneType
+        :param ignore_conditions:
+            Specify if pre and/or post conditions check is to be ignored for eligibility check.
+            The default is `IgnoreConditions.NONE`.
```
Comment (vyasr) [inline flow/project.py]:
```suggestion
            The default is :class:`~.IgnoreConditions.NONE`.
```

**Comment by vyasr** [comment]
@tcmoore3 once you review this feel free to merge if you approve.

**Pull request merged**